src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js

```diff
@@ -12,85 +13,92 @@ type Props = {
+                this.setDimensions();
+            }, DEBOUNCE_TIME)
+        );
 
-    @action setExpandedWidth = () => {
-        if (this.childRef.current && this.childRef.current.offsetWidth) {
-            const childWidth = this.childRef.current.offsetWidth;
-            if (this.showText) {
-                this.expandedWidth = childWidth;
-            } else {
-                if (childWidth > this.expandedWidth) {
-                    this.expandedWidth = childWidth;
-                }
-            }
+        if (!this.parentRef) {
+            return;
         }
 
-        this.setShowText();
-    };
+        resizeObserver.observe(this.parentRef);
+    }
+
+    @action
+    setDimensions = () => {
+        const {parentRef} = this;
+        const {childRef} = this;
 
-    @action setShowText = () => {
-        if (!this.parentRef.current || !this.parentRef.current.offsetWidth || !this.expandedWidth) {
+        if (childRef && childRef.offsetWidth > this.expandedWidth) {
+            this.expandedWidth = childRef.offsetWidth;
+        }
+
+        if (!parentRef) {
             return;
         }
 
-        const parentWidth = this.parentRef.current.offsetWidth;
+        this.parentWidth = parentRef.offsetWidth;
+    };
 
-        if (parentWidth >= this.expandedWidth) {
-            this.showText = true;
+    @computed
+    get showText(): boolean {
+        if (this.parentWidth >= this.expandedWidth) {
+            return true;
         } else {
-            this.showText = false;
+            return false;
         }
-    };
-
-    componentWillUnmount() {
-        window.removeEventListener('resize', this.setExpandedWidth);
     }
 
     render() {
-        const {
-            skin,
-            children,
-        } = this.props;
-
-        const itemsClass = classNames(
-            itemsStyles.items,
-            itemsStyles[skin]
-        );
+        const {skin, children} = this.props;
+
+        const itemsClass = classNames(itemsStyles.items, itemsStyles[skin]);
 
         return (
-            <div className={itemsStyles.itemsWrapper} ref={this.parentRef}>
-                <ul className={itemsClass} ref={this.childRef}>
-                    {children && React.Children.map(children, (item, index) => {
-                        return (
-                            <li key={index}>
-                                {React.cloneElement(item, {
-                                    ...item.props,
-                                    showText: this.showText,
-                                    skin: skin,
-                                })}
-                            </li>
-                        );
-                    })}
+            <div className={itemsStyles.itemsContainer} ref={this.setParentRef}>
+                <ul className={itemsClass} ref={this.setChildRef}>
+                    {children &&
+                        React.Children.map(children, (item, index) => {
+                            return (
```

Comment:
The return is actually not necessary if you write `(item, index) => (/* JSX */)`.